src/app/dataset-block/metadata-block/metadata-block.component.ts

```diff
@@ -58,9 +69,30 @@ export class MetadataBlockComponent
         );
     }
 
+    public onPageChange(params: {
+        currentPage: number;
+        isClick: boolean;
+    }): void {
+        this.loadHistory(params.currentPage - 1);
+    }
+
     private loadMetadataBlock(): Subscription {
         return this.blockService
             .requestMetadataBlock(this.datasetInfo, this.blockHash)
             .subscribe();
     }
+
+    private loadHistory(page = 0) {
+        return this.datasetService
```

Comment:
I wonder whether we need to subscribe to mediator's service event each time at all - maybe just separate the request initiation (each time) and subscription to the mediator's event (once)?

Reply:
Yes, refactored this method